/**
 *  Copyright (c) 2015 by Contributors
 */
syntax = "proto3";
package ntc;
option optimize_for = LITE_RUNTIME;
 
enum Type {
    TYPE_UNKNOWN = 0 ;
    SIGNUP  = 1 ;
    LOGIN   = 2 ;
    LOGOUT  = 3 ;
    MESSAGE = 4 ;   
    FILE    = 5 ;
    ERROR   = 6 ;
    
};

enum  Code {
    CODE_UNKNOWN     =      0;
    OK               =      1 ;
    REQUEST          =      2;
    SINGUP_EXEIST    =   3;
    SINGUP_SUCCESS   =   4;

    LOGIN_NOT_EXIST  =   5;
    LOGIN_WRONG_PASSWORD    = 6;
    LOGIN_SUCCESS    =   7;

    LOGOUT_SUCCESS   =   8;

    

};
 
// mete information about a message
message PBMETA {
    
    Type type = 1;
    Code code = 2;
    int32 dst = 3;
    int32 src = 4;
    repeated int32 data_size = 5;
}
 